packages/ns-api/files/ns.devices

```diff
@@ -0,0 +1,997 @@
+    if device_type == 'logical' and logical_type == 'bond':
+        values = get_bonding_values(
+            ip4_address, attached_devices, bonding_policy, bond_primary_device)
+    else:
+        # non-bond interfaces
+
+        values = {
+            'proto': protocol,
+            'ipaddr': '',
+            'netmask': '',
+            'ip6addr': '',
+            'gateway': '',
+            'ip6gw': '',
+            'username': '',
+            'password': '',
+            'clientid': '',
+            'vendorid': '',
+            'hostname': '',
+        }
+
+        if protocol == 'static':
+            # ip4 address and netmask
+            values['ipaddr'] = ip4_address
+
+            # ip6 address
+            if ip6_enabled and ip6_address:
+                values['ip6addr'] = ip6_address
+            else:
+                values['ip6addr'] = ''
+
+            if zone == 'wan':
+                # gateway
+                values['gateway'] = ip4_gateway
+
+                # ip6 gateway
+                if ip6_enabled:
+                    values['ip6gw'] = ip6_gateway
+                else:
+                    values['ip6gw'] = ''
+            else:
+                values['gateway'] = ''
+                values['ip6gw'] = ''
+        elif protocol == 'pppoe':
+            values['username'] = pppoe_username
+            values['password'] = pppoe_password
+        elif protocol in ['dhcp', 'dhcpv6']:
+            # dhcp client id
+            values['clientid'] = dhcp_client_id
+
+            if protocol == 'dhcp':
+                # dhcp vendor class
+                values['vendorid'] = dhcp_vendor_class
+
+                # dhcp hostname to send
+
+                dhcp_hostname = ''
+                delete_dhcp_hostname = False
+
+                match dhcp_hostname_to_send:
+                    case 'deviceHostname':
+                        delete_dhcp_hostname = True
+                    case 'doNotSendHostname':
+                        dhcp_hostname = '*'
+                    case 'customHostname':
+                        dhcp_hostname = dhcp_custom_hostname
+
+                if dhcp_hostname:
+                    values['hostname'] = dhcp_hostname
+
+                # delete dhcp hostname if needed
+                if interface_to_edit and delete_dhcp_hostname:
+                    uci.delete('network', interface_name, 'hostname')
+
+    # disable "force link" on red interfaces
+    if zone == 'wan':
+        values['force_link'] = '0'
+
+    for key, value in values.items():
+        uci.set('network', interface_name, key, value)
+
+
+def add_interface_to_firewall_zone(interface_name, zone, fw_zones, uci):
```

Comment:
Can we reuse functions from python3-nethsec lib? Like this one: https://nethserver.github.io/python3-nethsec/nethsec.firewall.html#add_interface_to_zone